File: Queue/Queue.java
// Queue in java

import java.util.Scanner;

class node
{
    public int data;
    public node next;

    public node(int no)
    {
        data = no;
        next =null;
    }
}

class QueueDemo
{
    public node head;

    public QueueDemo()
    {
        head = null;
    }

    public void enqueue(int no) // insertlast
    {
        node newn = new node(no);

        if(head == null)
        {
            head = newn;
        }
        else
        {
            node temp=head;

            while(temp.next != null)
            {
                temp= temp.next;
            }
            temp.next = newn;
        }
    }

    public void Dequeue() // deletefirst
    {
        if(head == null)
        {
            return ;
        }
        else if(head.next == null)
        {
            head = null;
            System.gc();
        }
        else
        {
            head = head.next;
            System.gc();
        }
    }

    public void Display()
    {
        node temp =head;

        while(temp.next != null)
        {
            System.out.println("|"+temp.data);
            temp = temp.next;
        }
    }
}

class Queue
{
    public static void main(String a[])
    {
        Scanner sc = new Scanner(System.in);

        QueueDemo qd = new QueueDemo();

        System.out.print("Insert Element in Queue:");
        qd.enqueue(50);
        qd.enqueue(40);
        qd.enqueue(30);
        qd.enqueue(20);
        qd.enqueue(10);
        qd.Display();
        System.out.println("\n----------------------------------------");

        System.out.print("\n\nDelete Element From Queue:");
        qd.Dequeue();
        qd.Display();
        System.out.println("\n----------------------------------------");
    }
}

File: Queue/Queue.c
// Queue 

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;
typedef struct node **PPNODE;

void Enqueue(PPNODE head,int no) // inserlast
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));
    newn->data =no;
    newn->next =NULL;

    if(*head == NULL)
    {
        *head = newn;
    }
    else
    {
        PNODE temp =*head;

        while(temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newn;
    }
}

void Dequeue(PPNODE head)
{
    if(*head == NULL)
    {
        return ;
    }
    else if((*head)->next ==NULL)
    {
        free(*head);
        *head = NULL;
    }
    else
    {
        PNODE temp = NULL;
        temp =*head;

        *head = (*head)->next ;
        free(temp);
    }
}

void Display(PNODE head)
{
    PNODE temp = head;

    while(temp->next != NULL)
    {
        printf("%d|",temp->data);
        temp =temp->next;
    }
}

int main()
{
    PNODE first =NULL;

    printf("Insert Element into Queue:\n");
    Enqueue(&first,80);
    Enqueue(&first,70);
    Enqueue(&first,60);
    Enqueue(&first,50);
    Enqueue(&first,40);
    Enqueue(&first,30);
    Enqueue(&first,20);
    Enqueue(&first,10);
    Display(first);
    printf("\n-------------------------------------------");

    printf("\n\nDelete Elements from queue:\n");
    Dequeue(&first);
    Dequeue(&first);
    Display(first);
    printf("\n--------------------------------------------");

    return 0;
}